modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -393,6 +410,12 @@ class Session implements ISession {
         // -- file porter config
         this.filePorter = new FilePorter(this)
 
+        if (config.cid) {
+            this.cidEnabled = true
+            this.cidStore = new DefaultCidStore()
```

Comment:
it's fine for now, but ideally the reference to this should be taken via the plugin factory

Reply:
Do you mean something similar to the ExecutorFactory? I could get from Plugins.manager the classes implementing the CidStore interface and add the `workflow.data.store.type` property to indicate the class.

Reply:
I think he means to use the same pattern as `CacheStore` and `CacheFactory` to load a CID backend through the plugin system

Reply:
Ok, I have used the same for the CidStore